node/src/actors/chain_manager/mining.rs

```diff
@@ -443,13 +440,19 @@ impl ChainManager {
                 // Collect outputs to be used as input for collateralized commitment,
                 // as well as outputs for change.
                 .and_then(move |vrf_proof, act, _| {
-                    let collateral_age = match &act.chain_state.chain_info {
-                        Some(x) => x.consensus_constants.collateral_age,
+                    let (collateral_age, checkpoint_period) = match &act.chain_state.chain_info {
+                        Some(x) => (x.consensus_constants.collateral_age, x.consensus_constants.checkpoints_period),
                         None => {
                             log::error!("ChainInfo is None");
                             return actix::fut::err(());
                         },
                     };
+
+                    // The timestamp included when using collateral is increased in (timeout - checkpoint_period)
+                    // to ensure that if your commit has not been accepted you can use your utxo in the next epoch
+                    // or at least in two epochs
+                    let ts_for_collateral = timestamp + tx_pending_timeout - u64::from(checkpoint_period) ;
```

Comment:
Is there any way to test this?

Reply:
The problem is that 'try_mine_data_request' is a ChainManager's impl function, so now we dont do tests between actors. So the way to test it is... creating a local testnet and check it :P

Reply:
Im trying and it is not working properly... u_u

Reply:
done!! At the end it was only change the timeout that we want, not the timestamp. And it could be tested